libs/auth/src/java_card.ts

```diff
@@ -255,6 +156,8 @@ export class JavaCard implements Card {
   }
 
   async checkPin(pin: string): Promise<CheckPinResponse> {
+    await this.selectApplet();
```

Comment:
It's not totally clear to me how frequently applet selection has to happen - I _think_ it's once per card insertion. Removing a card and re-inserting it does require the applet to be selected again.

So while `checkPin` doesn't explicitly require it (the applet will already have been selected in order to read the user), `programCard` definitely does. To play it safe, I'm just starting every public command with applet selection.